src/App.js

```diff
@@ -150,6 +152,50 @@ export default class App extends Component {
       })
   }
 
+  updateClass (klass) {
+    let newClass
+    if (this.state.activeCharacter.hasOwnProperty('klass')) {
+      newClass = Object.assign(this.state.activeCharacter, {klass: klass})
```

Comment:
this line mutates `this.state.activeCharacter`. If you want to avoid mutation (and you should), then you should be doing `Object.assign({}, this.state.activeCharacter, {klass: klass})`

However, I don't know the structure of `this.state.activeCharacter`, and if it's a deep object, then it won't necessarily fully copy.